java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -52,7 +65,81 @@
   public static final String KEY = "S2259";
   private static final RuleKey RULE_KEY = RuleKey.of(CheckList.REPOSITORY_KEY, KEY);
 
+  public enum AbstractValue {
+    // value is known to be not null.
+    NOTNULL,
+    // value is known to be null.
+    NULL,
+    // value is unknown (could be null or not null).
+    UNKNOWN
+  }
+
+  private class ConditionVisitor extends BaseTreeVisitor {
+    private State trueState;
+    @Nullable
+    private State falseState;
+
+    void visitCondition(ExpressionTree tree, State trueState, @Nullable State falseState) {
+      this.trueState = trueState;
+      this.falseState = falseState;
+      tree.accept(this);
+    }
+
+    @Override
+    public void visitBinaryExpression(BinaryExpressionTree tree) {
+      Symbol identifierSymbol;
+      if (tree.leftOperand().is(Tree.Kind.NULL_LITERAL) && tree.rightOperand().is(Tree.Kind.IDENTIFIER)) {
+        identifierSymbol = semanticModel.getReference((IdentifierTreeImpl) tree.rightOperand());
+      } else if (tree.leftOperand().is(Tree.Kind.IDENTIFIER) && tree.rightOperand().is(Tree.Kind.NULL_LITERAL)) {
+        identifierSymbol = semanticModel.getReference((IdentifierTreeImpl) tree.leftOperand());
+      } else {
+        return;
```

Comment:
Might worth to specify via the tests that a case like `if(a==null && b==null)` won't be covered.